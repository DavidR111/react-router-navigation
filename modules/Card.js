/* @flow */
/* eslint react/no-unused-prop-types: 0 */

import React, { Component, createElement } from 'react'
import { Match } from 'react-router'
import type { CardProps, CardState } from './CardTypeDefinitions'

type Props = CardProps & CardState

type MatchProps = {
  pattern: string,
  pathname: string,
  location: { pathname: string },
  isExact: boolean,
  pararms: Object,
}

type RendererProps = MatchProps & CardState

type State = {
  matchProps: ?MatchProps,
  pathname: ?string,
}

class Card extends Component<void, Props, State> {

  props: Props

  state: State = {
    pathname: null,
    matchProps: null,
  }

  static displayName = 'Card'

  renderMatchView = (matchProps: MatchProps): ?React$Element<RendererProps> => {
    // Set match props
    if (!this.state.pathname) this.state.pathname = matchProps.pathname
    if (
      this.state.pathname === matchProps.pathname ||
      !this.state.matchProps
    ) {
      this.state.matchProps = matchProps
    }
    // Render view with props
    const { render, component, isTransitioning, isFocused } = this.props
    const props = {
      ...this.state.matchProps,
      isTransitioning,
      isFocused,
    }
    if (render) return render(props)
    else if (component) return createElement(component, props)
    return null
  }

  shouldComponentUpdate() {
    return false
  }

  render(): React$Element<any> {
    return (
      <Match {...this.props}>
        {this.renderMatchView}
      </Match>
    )
  }

}

export default Card
