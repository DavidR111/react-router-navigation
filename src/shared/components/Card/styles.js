/* @flow */

import { StyleSheet } from 'react-native'

const styles = StyleSheet.create({
  container: {
    height: 300,
    backgroundColor: 'white',
  },
})

export default styles
