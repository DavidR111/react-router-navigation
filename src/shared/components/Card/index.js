/* @flow */

import React from 'react'
import { View, Text } from 'react-native'
import styles from './styles'

type Props = {
  children?: string,
}

const Card = (props: Props) => {
  const { children } = props
  return (
    <View style={styles.container}>
      <Text>{children}</Text>
    </View>
  )
}

export default Card
