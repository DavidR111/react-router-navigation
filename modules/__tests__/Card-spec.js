import React from 'react'
import { Text } from 'react-native'
import { Router } from 'react-router'
import createHistory from 'history/createMemoryHistory'
import renderer from 'react-test-renderer'
import './__mocks__'
import Card from './../Card'

it('<Card /> renders correctly', () => {
  const history = createHistory({
    initialEntries: ['/1'],
  })
  const component = renderer.create(
    <Router history={history}>
      <Card
        path="/:id"
        render={({ match: { params } }) => (
          <Text>{params.id}</Text>
        )}
      />
    </Router>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('<Card /> renders correctly with staticMatch prop when route doesn\'t match', () => {
  const history = createHistory({
    initialEntries: ['/foo/1'],
  })
  const component = renderer.create(
    <Router history={history}>
      <Card
        path="/foo/:id"
        render={({ staticMatch: { params } }) => (
          <Text>{params.id}</Text>
        )}
      />
    </Router>,
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
  history.push('/bar')
  tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
