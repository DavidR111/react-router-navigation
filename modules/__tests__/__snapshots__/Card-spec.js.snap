// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders correctly 1`] = `
<Text>
  1
</Text>
`;

exports[`<Card /> renders correctly with staticMatch prop when route doesn't match 1`] = `
<Text>
  1
</Text>
`;

exports[`<Card /> renders correctly with staticMatch prop when route doesn't match 2`] = `
<Text>
  1
</Text>
`;
